internal/controller/controllers/hypershiftagentserviceconfig_controller.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	_ "embed"
+
+	aiv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+	logutil "github.com/openshift/assisted-service/pkg/log"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// AgentServiceConfigReconciler reconciles a AgentServiceConfig object
+type HypershiftAgentServiceConfigReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Log    logrus.FieldLogger
+
+	// Namespace the operator is running in
+	Namespace string
```

Comment:
I'm not sure why [this conversation](https://github.com/openshift/assisted-service/pull/4381#discussion_r969986487) got marked as outdated, but it still isn't clear to me that this is needed and it's not used in this PR, so please remove it until we have a use for it.

Reply:
Done.